frontend/src/TooljetDatabase/Table/index.jsx

```diff
@@ -89,6 +89,80 @@ const Table = ({ collapseSidebar }) => {
   const [isCellUpdateInProgress, setIsCellUpdateInProgress] = useState(false);
   const [referencedColumnDetails, setReferencedColumnDetails] = useState([]);
 
+  const [cachedOptions, setCahedOptions] = useState([]);
+
+  const fetchFkDataForColumn = (currentColumn, currentIndex) => {
+    // responsible to fetch the fk column details for all available fk's and cache them
+    return new Promise((resolve, reject) => {
+      if (!currentColumn?.referenced_column_names?.length) return resolve();
```

Comment:
Rather than checking `currentColumn?.referenced_table_id,` on line 104 should we check here